pkg/ingestor/parser/spdx/parse_spdx.go

```diff
@@ -0,0 +1,243 @@
+}
+
+func (s *spdxParser) getFileElement(elementID string) []assembler.ArtifactNode {
+	if fileNode, ok := s.files[string(elementID)]; ok {
+		return fileNode
+	}
+	return nil
+}
+
+func (s *spdxParser) CreateEdges(foundIdentities []assembler.IdentityNode) []assembler.GuacEdge {
+	edges := []assembler.GuacEdge{}
+	toplevel := s.getPackageElement("SPDXRef-DOCUMENT")
+	// adding top level package edge manually for all depends on package
+	if toplevel != nil {
+		edges = append(edges, createTopLevelEdges(toplevel[0], s.packages)...)
+	}
+	for _, rel := range s.spdxDoc.Relationships {
+		foundPackNodes := s.getPackageElement("SPDXRef-" + string(rel.RefA.ElementRefID))
+		foundFileNodes := s.getFileElement("SPDXRef-" + string(rel.RefA.ElementRefID))
+		relatedPackNodes := s.getPackageElement("SPDXRef-" + string(rel.RefB.ElementRefID))
+		relatedFileNodes := s.getFileElement("SPDXRef-" + string(rel.RefB.ElementRefID))
+		if len(foundPackNodes) > 0 {
+			for _, packNode := range foundPackNodes {
+				createdEdge := sort(packNode, rel.Relationship, relatedPackNodes, relatedFileNodes)
+				if createdEdge != nil {
+					edges = append(edges, createdEdge)
+				}
+			}
+		}
+		if len(foundFileNodes) > 0 {
+			for _, fileNode := range foundFileNodes {
+				createdEdge := sort(fileNode, rel.Relationship, relatedPackNodes, relatedFileNodes)
+				if createdEdge != nil {
+					edges = append(edges, createdEdge)
+				}
+			}
+		}
+	}
+	return edges
+}
+
+func createTopLevelEdges(toplevel assembler.PackageNode, packages map[string][]assembler.PackageNode) []assembler.GuacEdge {
+	edges := []assembler.GuacEdge{}
+	for _, packNodes := range packages {
+		for _, packNode := range packNodes {
+			if packNode.Purl != toplevel.Purl {
+				e := assembler.DependsOnEdge{
+					PackageNode:       toplevel,
+					PackageDependency: packNode,
+				}
+				edges = append(edges, e)
+			}
+		}
+	}
+	return edges
+}
+
+func sort(foundNode assembler.GuacNode, relationship string, relatedPackNodes []assembler.PackageNode, relatedFileNodes []assembler.ArtifactNode) assembler.GuacEdge {
+	if len(relatedFileNodes) > 0 {
+		for _, rfileNode := range relatedFileNodes {
+			return sortByType(relationship, foundNode, rfileNode)
+		}
+	} else if len(relatedPackNodes) > 0 {
+		for _, rpackNode := range relatedPackNodes {
+			return sortByType(relationship, foundNode, rpackNode)
+		}
+	}
+	return nil
+}
+
+func sortByType(relationship string, foundNode assembler.GuacNode, relatedNode assembler.GuacNode) assembler.GuacEdge {
+	switch relationship {
+	case spdx_common.TypeRelationshipContains:
+		return sortContainsEdge(foundNode, relatedNode)
+	case spdx_common.TypeRelationshipDependsOn:
+		return sortDependsOnEdge(foundNode, relatedNode)
+	}
+	return nil
+}
+
+func sortContainsEdge(foundNode assembler.GuacNode, relatedNode assembler.GuacNode) assembler.GuacEdge {
+	e := assembler.ContainsEdge{}
+	if foundNode.Type() == "Package" {
+		e.PackageNode = foundNode.(assembler.PackageNode)
+	}
```

Comment:
this hsould err if this is not the case, there is no alternative path and will lead to a misdefined edge